src/backend/distributed/commands/multi_copy.c

```diff
@@ -1421,22 +1684,31 @@ AppendCopyBinaryFooters(CopyOutState footerOutputState)
  */
 static void
 StartCopyToNewShard(ShardConnections *shardConnections, Oid relationId,
-					CopyStmt *copyStatement)
+					CopyStmt *copyStatement, PGconn *masterConnection)
```

Comment:
masterConnection seems to be such a global thing, I wonder whether we should just make it a global variable.